### PR TITLE
refactor(ts-wrapped): simplify metadata writing

### PR DESCRIPTION
**Pull request by alfaproject**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?**
`tsc-wrapper` is checking all `js` source files to write the metadata one, and that seems a bit superfluous.

**What is the new behavior?**
The referenced ticket from TS was released a long time ago, and I guess this behaviour can be simplified by writing the metadata file when the typings are being written.

**Does this PR introduce a breaking change?**
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: tools/@angular/tsc-wrapped/src/compiler_host.ts [hunk 1]
```diff
@@ -73,54 +73,33 @@ interface DecoratorInvocation {
...
-          return;
+          // Write the metadata file alongside type definitions
+          this.writeMetadata(fileName, sourceFiles[0]);
         }
-
-        if (IGNORED_FILES.test(fileName)) {
-          return;
-        }
-
-        if (!sourceFiles) {
```
Comment (chuckjaz) [inline tools/@angular/tsc-wrapped/src/compiler_host.ts]: These checks are still required for 2.4.

**Review by chuckjaz — CHANGES_REQUESTED** [review]
Please rebase the changes.

**Comment by tbosch** [comment]
This is a change we are willing to take in the next release. Could you please rebase against master if you are still interested in landing this PR?

**Comment by vicb** [comment]
Closing as outdated

**Pull request closed without merging**